### ISSUE TITLE
Change starting keep?

### ISSUE THREAD

**Issue by ckolar612** [Sh0wdown/UnofficialCrusaderPatch#181]
Hey there!  Just a quick idea.

Is it possible to make it so that, in multiplayer, everyone (or at least human lords) can start with a different keep?  It would be fun to play a skirmish where you have the large Stronghold, as oppose to the basic square bit.

Also, great job on the updates thus far!  This is breathing new life into the game for me.

**Comment by PitchNeeded** [comment]
I would like the option for the AIs to start with the biggest keep (as well as looking visually great, it might even make their last stand defense better), but it might cause errors in building their castle :(

**Comment by Monsterfisch** [comment]
while it would look cool the problem here is that the different keeps have a different footprint size meaning all the castle layouts would have to be adjusted to the bigger keeps entailing an update to the aiv editor as well. So I would say its not quite worth the effect for the amount of rework would be required to make it viable

**Comment by Heroesflorian** [comment]
I have my doubts this would be feasible, though. 

First of all, the larger, higher keep would be an unfair defensive advantage that would be impossible to really balance out. 
And the other keep that has the same size as the "default" one, but cannot be entered by Lord or units, would be strictly inferior compared to the default keep, so why would anyone choose it at all? (Plus, this one could screw up the AI behaviour even if it fits the AIVs in terms of keep size!) 
Unlike for SH1 (where you had e.g. wooden keep and stone keep, both of same size and capabilities, but different visual style), there are no similar keep variants in SHC that could be exchanged based on player preference without nasty complications. 

Second, unlike for "place your own keep" scenario maps, skirmish maps have fixed keep locations. Replacing the existing keeps on a map with a different keep that is two times bigger is asking for all kinds of issues... e.g. bigger keeps standing half on top of a cliff, half below it, bigger keeps conflicting with water, bigger keeps near map edges possibly bugging out and disappearing - this can happen already for AIVs where the keep has been relocated to a less centered position, and can cause the keep along with peasants and the player's Lord to disappear altogether, so larger keeps could be in similar troubles on certain maps -, bigger keeps blocking access paths to / from certain parts of the map, ...

To sum it up: Hard/impossible to balance, unusable for AI players, of quite limited use for human players, and burdened with a lot of tough issues...

**Comment by Monsterfisch** [comment]
only saving grace I could see would be a different looking keep design with the same footprint if someone figures out a way to get extra files in there and actually loaded via an interface that way we could have a European keep and an Arabian one. But that sounds a lot like something worth looking into at a much later point in time, you know after gameplay stuff is as repaired as possible.

**Comment by Heroesflorian** [comment]
Yes, that kind of stuff - i.e. visual customization without gameplay changes - would be neat for sure! 

Be it Arabian keeps, mosques instead of churches, "Arabian stuff" instead of ale, or perhaps even visually differing styles of walls/crenellations (in SHC2, this actually is a thing, see below) as well as gates and towers. 

Not gonna happen anytime soon, though, considering the difficulties of getting the files in, getting them to work through patching and the long list of other feature requests. 

![grafik](https://user-images.githubusercontent.com/12549394/51442645-342a8f80-1cdf-11e9-9b7e-204660492fc2.png)

![grafik](https://user-images.githubusercontent.com/12549394/51442650-499fb980-1cdf-11e9-8ad4-26c42286e31f.png)

**Comment by Krarilotus** [comment]
As this is not viable to implement with how the game is currently functioning (keep placements on terrain would be blocked if rocks are nearby, AI's wont function with different keep sizes etc.) I dont see how this would be worked on. If you disagree or these things change, feel free to reopen this issue.
